CMakeLists.txt

```diff
@@ -40,6 +40,7 @@ option(ENABLE_STATIC_LIBRARY_MODULES "If ON builds Valhalla modules as STATIC li
 set(LOGGING_LEVEL "" CACHE STRING "Logging level, default is INFO")
 set_property(CACHE LOGGING_LEVEL PROPERTY STRINGS "NONE;ALL;ERROR;WARN;INFO;DEBUG;TRACE")
 set_property(GLOBAL PROPERTY USE_FOLDERS ON)
+include_directories(SYSTEM ${VALHALLA_SOURCE_DIR}/third_party/date/include)
```

Comment:
why move this out to the top level? design-wise we were really hoping to get all date functionality wrapped by baldr (in helper functions mostly).

a little background on this, a long time ago (https://github.com/valhalla/valhalla/pull/1512/files#r215660149) we started using the date library but it has, on different platforms, different ways of getting at timezone information. for us, the timezones have to be in a stable order (because we use indices to reference them from the data that we build). this is why tz_alt exists. i copied tz.cpp from the date library and modified it to use static data. because this code isnt owned by us (nor should it be really) i dont think we need to bother about fixing warnings. in fact i'd really have prefered that we make as little changes to that source as possible so it was clear what we did there vs what comes standard in that thirdparty library.

Reply:
Treating the third party header as system header, the compiler wont generate warnings from that header.
By having it at top level, will help me out on fixing the other modules

Reply:
i think we should patch the valhalla module function to work with the SYSTEM header directive then we can keep this inclusion inside of baldr with the rest of the date time stuff

Reply:
Added a task on the #4018 will work once things compile without warnings